components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/template/APITemplateBuilderImpl.java

```diff
@@ -109,14 +152,57 @@ public void addHandler(String handlerName, Map<String, String> properties) {
 		addHandlerPriority(handlerName, properties, handlers.size());
 	}
 
-	public void addHandlerPriority(String handlerName, Map<String, String> properties, int priority) {
+	public void addHandlerPriority(String handlerName, Map<String, String> properties,
+								   int priority) {
 		HandlerConfig handler = new HandlerConfig(handlerName, properties);
 		handlers.add(priority, handler);
 	}
 
-	public String getTemplatePath() {
+	private String getVersionedWebAppTemplatePath() {
+		return "repository" + File.separator + "resources" + File.separator + "api_templates" +
+				File.separator +
+				APITemplateBuilderImpl.VELOCITY_TEMPLATE_SYNAPSE_CONFIG_VERSIONED_WEBAPP;
+	}
+
+	private String getNonVersionedWebAppTemplatePath() {
 		return "repository" + File.separator + "resources" + File.separator + "api_templates" +
-		       File.separator + APITemplateBuilderImpl.TEMPLATE_TYPE_VELOCITY + ".xml";
+				File.separator +
+				APITemplateBuilderImpl.VELOCITY_TEMPLATE_SYNAPSE_CONFIG_NON_VERSIONED_WEBAPP;
 	}
 
+	private String getVelocityLoggerName() {
+		AppManagerConfigurationService config =
+				ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService();
+		String velocityLogPath = config.getAPIManagerConfiguration().getFirstProperty(
+				AppMConstants.VELOCITY_LOGGER);
+		if (velocityLogPath != null && velocityLogPath.length() > 1) {
+			return velocityLogPath;
+		} else {
+			return null;
+		}
+	}
+
+	private String processTemplate(VelocityEngine ve, VelocityContext vc, String templatePath)
```

Comment:
Short variable names were chosen in order to not to overflow max line size (at the time this code snippet was committed the max line size was 100).
fixed in commit 9d756e1d1fcad8ab8bcec5281405fa0645fa75d9 of PR #153